velox/exec/tests/utils/ExpressionBuilder.h

```diff
@@ -0,0 +1,311 @@
+///
+/// > ExprPtr e = isNull("col"_f) // "col is null"
+/// > ExprPtr e = !isNull("col"_f) // "col is not null"
+///
+/// Conjuncts and "between":
+///
+/// > ExprPtr e = "a"_f && "b"_f || "c"_f; // "a AND b OR c"
+/// > ExprPtr e = between("a"_f, 0L, 10L); // "a between 0 and 10"
+///
+/// C++ literals are automatically converted into ConstantExpr (expression
+/// literals) when part of an expression. To explicitly create a literal:
+///
+/// > ExprPtr e = literal(10.3);
+/// > ExprPtr e = literal("str");
+/// > ExprPtr e = literalTimestamp("1970-01-01 01:30:00");
+///
+/// Arithmetic operators are also overloaded:
+///
+/// > ExprPtr e = "col"_f + 1 - 10; // "col + 1 - 10"
+///
+/// In any expression, as long as one of the sides is an expression node, the
+/// correct expression will be created. For example, both version work as
+/// expected:
+///
+/// > ExprPtr e = "col"_f * 100; // "col * 100"
+/// > ExprPtr e = 100 * "col"_f; // "100 * col"
+///
+/// When building long expressions, be careful about C++ operator precedence.
+/// For example:
+///
+/// > ExprPtr e = "col"_f + 5 * 100;
+///
+/// will generate the expression "col + 500" since C++ will first sum the two
+/// integers. To force the expected behavior, you can explicitly spell out the
+/// literal:
+///
+/// > ExprPtr e = "col"_f + 5 * literal(100);
+/// > ExprPtr e = "col"_f + literal(5) * 100;
+///
+/// Both will generate "col + 5 * 100", which is "plus(col, multiply(5, 100))".
+///
+/// Generic function calls can be created using `call()`:
+///
+/// > ExprPtr e = call("func", 10L); // "func(10)"
+///
+/// `call()` supports arbitrary parameters, which can be other expressions or
+/// (C++) literals.
+///
+/// Aliases can be defined using `alias()`:
+///
+/// > ExprPtr e = alias("col"_f, "foo"); // "col as foo"
+/// > ExprPtr e = alias(10, "foo"); // "10 as foo"
+///
+/// All functions above can be nested and combined in arbitrary ways.
+///
+/// > ExprPtr e = 10L * "c1"_f > call("func", 3.4, "g"_f / "h"_f, call("j"));
+///
+/// is the same as "10 * c1 > func(3.4, g / h, j())".
+
+using core::CallExpr;
+using core::ConstantExpr;
+using core::ExprPtr;
+using core::FieldAccessExpr;
+using core::IExpr;
+
+/// Wrapper library used so we can safely overload operators.
+struct ExprWrapper {
+  std::shared_ptr<IExpr> ptr;
+
+  // Enable proper comparisons.
+  bool operator==(const ExprWrapper& other) const {
+    return *ptr == *other.ptr;
+  }
+
+  bool operator==(const ExprPtr& other) const {
+    return *ptr == *other;
+  }
+
+  // Provide better gtest failure messages.
+  friend std::ostream& operator<<(std::ostream& os, const ExprWrapper& expr) {
+    return os << expr.ptr->toString();
+  }
+
+  // For convenience, enable implicit conversions to ExprPtr.
+  operator ExprPtr() const {
+    return ptr;
+  }
+};
+
+/// Field access (references to columns).
+inline ExprWrapper field(const std::string& name) {
+  return {std::make_shared<FieldAccessExpr>(name, std::nullopt)};
```

Comment:
The field access could contain 'input', which makes it a dereference expression. Do we need to support it? Thanks.